### DISCUSSION TITLE
LS-QVL installs, reboots then only option is install again

### DISCUSSION THREAD

**Discussion by Darkseid51** [1000001101000/Debian_on_Buffalo#138] [Q&A] [answered]
Hi,

TLDR; Debian installs, reboots on finish then only access is via ssh as installer@device which kicks off install again in a loop.

I'm unable to login as root or user account setup during installation. Hopefully I'm missing something simple...

Long version:
I've tried options 1 and 2 in the following
https://github.com/1000001101000/Debian_on_Buffalo/wiki/Preparing-Device-for-Debian-Install

And setup partitions as per referenced video at 
https://www.youtube.com/watch?v=dfMHlLmsSF8

Based on video above, partition table was setup as follows:

RAID1 device #0 - 1.0 GB Linux Software RAID Array 
#1             1.0 GB    F  ext3                 /boot
RAID1 device #1 - 5.1 GB Linux Software RAID Array
#1             5.1 GB    F  ext3                 /
RAID1 device #10 - 1.0 GB Linux Software RAID Array
#1             1.0 GB    F  swap                 swap
RAID1 device #2 - 2.0 TB Linux Software RAID Array
#1  primary    2.0 TB    f  ext4                 /mnt/array2
SCSI1 (2,0,0) (sda) - 2.0 TB ATA SAMSUNG HD204UI
               1.0 MB       FREE SPACE
#1             1.0 GB       ext3        primary
#2             5.1 GB                   primary
#3             1.0 MB                   primary
#4             1.0 MB                   primary
#5             1.0 GB                   primary
#6             2.0 TB                   primary
#7           255.9 MB       ext2
#8             7.7 GB       ext4
(truncated prior to second 2TB SCSI physical drive)
-=-

dmesg (when logged in as installer@device immediately after install/reboot)

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.19.0-20-marvell (debian-kernel@lists.debian.org) (gcc version 8.3.0 (Debian 8.3.0-6)) #1 Debian 4.19.235-1 (2022-03-17)
[    0.000000] CPU: Feroceon 88FR131 [56251311] revision 1 (ARMv5TE), cr=0005397f
[    0.000000] CPU: VIVT data cache, VIVT instruction cache
[    0.000000] OF: fdt: Machine model: Buffalo Linkstation LS-QVL
[    0.000000] Memory policy: Data cache writeback
[    0.000000] On node 0 totalpages: 65536
[    0.000000]   Normal zone: 512 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 65536 pages, LIFO batch:15
[    0.000000] random: get_random_bytes called from start_kernel+0x78/0x4ac with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 65024
[    0.000000] Kernel command line: console=ttyS0,115200 root=/dev/sda2 rw initrd=0x00800040,12M panic=5 BOOTVER=0.26 mtdparts=nand_mtd:0x1000000(boot),0x1e800000(rootfs),0x800000(reserve)
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Memory: 241424K/262144K available (3992K kernel code, 342K rwdata, 1040K rodata, 312K init, 219K bss, 20720K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] Virtual kernel memory layout:
...
[    2.036472] 0x000000000000-0x000000060000 : "uboot"
[    2.042906] orion-ehci f1050000.ehci: EHCI Host Controller
[    2.050829] 0x000000060000-0x000000070000 : "dtb"
[    2.057892] orion-ehci f1050000.ehci: new USB bus registered, assigned bus number 1
[    2.071547] 0x000000070000-0x000000080000 : "uboot_env"
[    2.083248] orion-ehci f1050000.ehci: irq 30, io mem 0xf1050000
[    2.105816] orion-ehci f1050000.ehci: USB 2.0 started, EHCI 1.00
[    2.112049] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 4.19
[    2.120367] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.127631] usb usb1: Product: EHCI Host Controller
[    2.132539] usb usb1: Manufacturer: Linux 4.19.0-20-marvell ehci_hcd
[    2.138930] usb usb1: SerialNumber: f1050000.ehci
[    2.145163] hub 1-0:1.0: USB hub found
[    2.148992] hub 1-0:1.0: 1 port detected
[    2.405810] usb 1-1: new high-speed USB device number 2 using orion-ehci
[    2.505819] ata1: SATA link up 3.0 Gbps (SStatus 123 SControl F300)
[    2.512791] ata1.15: Port Multiplier 1.2, 0x11ab:0x4140 r193, 4 ports, feat 0x5/0x1f
[    2.530194] ata1.00: hard resetting link
[    2.566249] usb 1-1: New USB device found, idVendor=0424, idProduct=2512, bcdDevice= b.b3
[    2.574488] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    2.583376] hub 1-1:1.0: USB hub found
[    2.587363] hub 1-1:1.0: 2 ports detected
[    2.854852] ata1.01: hard resetting link
[    3.174584] ata1.02: hard resetting link
[    3.654058] ata1.03: hard resetting link
[    4.154019] ata1.02: ATA-8: SAMSUNG HD204UI, 1AQ10001, max UDMA/133
[    4.160329] ata1.02: 3907029168 sectors, multi 0: LBA48 NCQ (depth 31/32)
[    4.186013] ata1.02: configured for UDMA/133
[    4.210003] ata1.03: ATA-8: SAMSUNG HD204UI, 1AQ10001, max UDMA/133
[    4.216310] ata1.03: 3907029168 sectors, multi 0: LBA48 NCQ (depth 31/32)
[    4.242010] ata1.03: configured for UDMA/133
[    4.246486] ata1: EH complete
[    4.249762] scsi 0:2:0:0: Direct-Access     ATA      SAMSUNG HD204UI  0001 PQ: 0 ANSI: 5
[    4.260593] scsi 0:3:0:0: Direct-Access     ATA      SAMSUNG HD204UI  0001 PQ: 0 ANSI: 5
[    4.584070] ata2: SATA link down (SStatus 0 SControl F300)
[    4.675619] sd 0:2:0:0: [sda] 3907029168 512-byte logical blocks: (2.00 TB/1.82 TiB)
[    4.684883] sd 0:3:0:0: [sdb] 3907029168 512-byte logical blocks: (2.00 TB/1.82 TiB)
[    4.703111] sd 0:2:0:0: [sda] Write Protect is off
[    4.707961] sd 0:2:0:0: [sda] Mode Sense: 00 3a 00 00
[    4.708018] sd 0:3:0:0: [sdb] Write Protect is off
[    4.712848] sd 0:3:0:0: [sdb] Mode Sense: 00 3a 00 00
[    4.712916] sd 0:2:0:0: [sda] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    4.722085] sd 0:3:0:0: [sdb] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    4.765142]  sda: sda1 sda2 sda3 sda4 sda5 sda6 sda7 sda8 sda9
[    4.776335] sd 0:2:0:0: [sda] Attached SCSI disk
[    4.783083]  sdb: sdb1 sdb2 sdb3 sdb4 sdb5 sdb6 sdb7 sdb8 sdb9
[    4.797601] sd 0:3:0:0: [sdb] Attached SCSI disk
[   12.370600] random: crng init done
[   15.762082] mv643xx_eth_port mv643xx_eth_port.0 eth0: link up, 1000 Mb/s, full duplex, flow control enabled
[   15.762180] mv643xx_eth_port mv643xx_eth_port.0 eth0: link down
[   17.178884] NET: Registered protocol family 17
[   19.060480] mv643xx_eth_port mv643xx_eth_port.0 eth0: link up, 1000 Mb/s, full duplex, flow control disabled

Any tips appreciated. Have tried variety of LVM, manual partition, whole disk etc

Re,
Chris

**Comment by 1000001101000** [comment]
Good to hear from you!

if it’s booting back into the installer then the boot files aren’t getting wiped out by formatting then replaced by the install as expected.

I think a next step would be to try again and post a screenshot of the partitioning menu before moving forward so I can take a look.

**Comment by Darkseid51** [comment]
Hi,

Thanks for the prompt reply.
Just solved.

After trying lots of partition combinations I eventually ended up with the
NAS in emergency mode.
To fix I formatted the drives in Windows to single NTFS partition, then
reinstalled into the NAS
Using LinkStation Updater to reinstall factory firmware and partitioning.

Transferred boot files with acp commander, repartitioned as before (boot,
root, swap)
This time messing around with advanced mode I selected (sorry forget exact
name) make bootable menu item, then went back to installing packages.
I've seen a fair few run throughs and seems to previously have gone from
installing packages to reboot, without the making bootable tasks appearing
on screen.

Wasn't a precise process but hopefully the above helps someone else muddle
through if they have the same issue.

Re,
Chris

**Comment by 1000001101000** [comment, marked as answer]
I think it boils down to "started over from fresh stock firmware install and it worked". Though having a big NTFS partition likely caused the disk detection to fail entirely and force it to NAND boot into EM mode. That could be a handy trick, though only for devices that have NAND to boot from I imagine. Glad to hear you got it working!

I can think of a few reasons it got to that state, high on the list being the old /boot array got out of sync and the device kept choosing one with the old boot files etc.

I should have probably mentioned earlier, I should have the Bullseye installer up for this device in the next few days if you want to try that out when the time comes.

**Reply by Darkseid51** [reply]
I originally tried removing all partitions, which triggered emergency mode
The stock updater was unable to partition, so I plugged them into the PC, formatted as single 2Tb NTFS and that the firmware updater seemed happy to nuke.

I'm redoing the factory firmware now as
1. interested to see if the make bootable issue occurs again, now factory disk layout, if choose non advanced install
2. will then put in an extra couple of data disks I want to retain and see if I can access/backup

Quite possibly I'll still be trying things for a while so should be able check out the Bullseye installer

**Comment by 1000001101000** [comment]
feel free to post your partitioning screen when you get there, I can take a look. 

the bootable flag shouldn't be a factor, at least it hasn't been for the previous few hundred installs I'm aware of.  The presence of disks with a 2tb "partition #1" can be a factor, but only a negative one.

**Reply by Darkseid51** [reply]
Where I was at:
Working Debian install on a pair of 2Tb drives in bays 3 & 4.
I reinserted a pair of 3Tb drives with data I wanted to retain and QVL insisted on booting from those, regardless of which bays used.

To 'fix' I mounted the 3Tb drives on another system and removed the stock boot files.
Restored QVL to stock firmware with just the dual 2Tb drives (letting it partition them) then inserted the 3Tb drives and used stock OS to access and backup the 3Tb drive content (this time).

Followed the steps from scratch and installed Debian without issue - this time with both 2Tb and 3Tb pairs inserted.
Moral of the story, have a backup, go from stock.

Before install (may have typos took screenshot drag and paste didn't want to play)

RAID1 device  #0 - 1.0 GB Linux Software RAID Array
> #1 1.0 GB ext3
RAID1 device #1 - 5.1 GB Linux Software RAID Array
> #1 5.1 GB ext3
RAID1 device #10 - 1.0 GB Linux Software RAID Array
> #1 1.0 GB F swap swap
RAID1 device #2 - 2.0 TB Linux Software RAID Array
> #1 2.0 TB xfs
RAID1 device #3 - 3.0 TB Linux Software RAID Array
> #1 3.0 TB xfs
SCSI1 (0,0,0) (sda) - 3.0 TB ATA WDC WD30EFRX-68E
> 1.0 MB FREE SPACE
> #1 1.0 GB ext3 primary
> #2 5.1 GB primary
> #3 1.0 MB primary
> #4 1.0 MB primary
> #5 1.0 GB primary
> #6 3.0 TB primary
etc for other disks

I changed the first couple of partitions to
RAID1 device  #0 - 1.0 GB Linux Software RAID Array
> #1 1.0 GB ext3 /boot
RAID1 device #1 - 5.1 GB Linux Software RAID Array
> #1 5.1 GB ext3 /

After install:

parted /dev/md0

Model: Linux Software RAID Array (md)
Disk /dev/md0: 1024MB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 
Number  Start  End     Size    File system  Flags
 1      0.00B  1024MB  1024MB  ext3

parted /dev/md1

Model: Linux Software RAID Array (md)
Disk /dev/md1: 5119MB
Sector size (logical/physical): 512B/4096B
Partition Table: loop
Disk Flags: 
Number  Start  End     Size    File system  Flags
 1      0.00B  5119MB  5119MB  ext3

parted /dev/sda

Model: ATA WDC WD30EFRX-68E (scsi)
Disk /dev/sda: 3001GB
Sector size (logical/physical): 512B/4096B
Partition Table: gpt
Disk Flags: 
Number  Start   End     Size    File system  Name     Flags
 1      1049kB  1026MB  1024MB  ext3         primary  msftdata
 2      1026MB  6146MB  5120MB               primary  msftdata
 3      6146MB  6147MB  1049kB               primary  msftdata
 4      6147MB  6148MB  1049kB               primary  msftdata
 5      6148MB  7172MB  1024MB               primary  msftdata
 6      7172MB  2992GB  2985GB               primary  msftdata

sdb is the same as above
sdd is the same as below

Model: ATA SAMSUNG HD204UI (scsi)
Disk /dev/sdc: 2000GB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 
Number  Start   End     Size    File system  Name     Flags
 1      1049kB  1026MB  1024MB  ext3         primary  msftdata
 2      1026MB  6146MB  5120MB               primary  msftdata
 3      6146MB  6147MB  1049kB               primary  msftdata
 4      6147MB  6148MB  1049kB               primary  msftdata
 5      6148MB  7172MB  1024MB               primary  msftdata
 6      7172MB  1992GB  1985GB               primary  msftdata

Now to sort out the buttons and full speed fan...

Cheers,
Chris

**Comment by 1000001101000** [comment]
That makes some sense. 
The multi-bay devices typically scan the "partition 1" of each disk and then picks one to boot from (note it's the partition labeled `#1` not the first partition). This can cause failures if one of those partitions is too big for it to read since the bootloader gives up/crashes before it tries booting anything. It also checks the timestamp on the boot files and looks for a special signature in the partition table which is set by the stock firmware install, it sounds like this model prefers disks with that signature over ones that don't.

I haven't spent too much looking at that aspect of the boot process other than to notice that it is there:
https://buffalonas.miraheze.org/wiki/Terastation_TS-XL#Partition_table_notes

I could probably add some utility to the installer to set that value and/or clear it etc but that would likely be unreliable since other tools like fdisk could just overwrite and would risk side effects etc. 

You can prevent that issue from happening a number of ways:

- remove the first partition on the disks you don't want to boot from
- renumber the first partition to something other than 1 on those disks
- remove the signature (both of the above probably also do that)
- remove the filesysten and/or boot files from those partitions (I think that's what you did)
